### PR TITLE
Update the README and env files to more accurately reflect the setup process.

### PR DESCRIPTION
**Pull request by nasainsbury**
When trying to setup and test R=, I noticed a few minor things that stopped me, which weren't immediately apparent in the documentation. This PR will attempt to address some of those issues.

Some of what I've done may be wrong, so please bare with me whilst I get used to the context.

Here were a few things I noticed:

1. The application needs to be running `node 16`, I was running `node 14` and was getting errors. I've updated my `node` version along with the README to reflect this.
2. It's not immediately apparent what env variables are needed to run locally, I've tried to address that in the `.env.example` file.
3. I've added a `docker-compose.yml` file so it's easier to get postgres setup locally, this will likely be added to to replicate some of the services locally.
4. To commit to the repo (even in a PR), pre-commit tests are being run, these require you to have a postmark account. However, can be difficult to set one up if you don't have your own domain. One idea to get round this would be to run the tests via github actions after the commit, but before the merge?

I've removed the email testing as it requires postmark account, well I've skipped it for now.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
 CROSSREF_LOGIN_PASSWD
 CROSSREF_URL
-DATABASE_URL
+
+# Postgres DB URL. Default values found in ./docker-compose.yml
+# This is needed to do any local development
+DATABASE_URL="postgres://dbuser:dbpassword@localhost:5432/researchequals-dev"
+
+
+# Used for connecting Ocrid account to user. Can setup credentials for free by creating an account: https://orcid.org/ and looking at developer settings
```
Comment (chartgerink) [inline .env.example:14]:
```suggestion
# Used for connecting ORCID account to user. Can setup credentials for free by creating an account: https://orcid.org/ and looking at developer settings
```

File: .env.example [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
 UPLOADCARE_SECRET_KEY
+
+# Freemium software, used for free text searching, https://https://www.algolia.com/
 ALGOLIA_APP_ID
 ALGOLIA_API_SEARCH_KEY
 ALGOLIA_API_ADMIN_KEY
 ALGOLIA_PREFIX
+
+# Used for cronjobs, https://quirrel.dev/
+# This is not needed for local development
```
Comment (chartgerink) [inline .env.example:35]:
```suggestion
# Used for cronjobs and queues
# https://github.com/quirrel-dev/quirrel
# This is not needed for local development
```

The service itself is discontinued so you need to self-deploy this for production 😿

File: .env.example [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
 QUIRREL_TOKEN
 QUIRREL_BASE_URL
 QUIRREL_ENCRYPTION_SECRET
+
+# Upon publication of a module, a DOI is created using CROSSREF (envs above), the DOI PREFIX is set by crossref for public
+# This is only needed if you are testing the publication of modules
```
Comment (chartgerink) [inline .env.example:41]:
```suggestion
# Upon publication of a module, a DOI is created using CrossRef (envs above)
# You must use the prefix issued to you by CrossRef in both testing and production
# This is only needed if you are testing the publication of modules
```

**Review by chartgerink — APPROVED** [review]
Cheers @nasainsbury! Thanks a bunch for these suggestions and improvements.  🥳 

This will help onboard new contributors. Hope to see you around on this repo, and feel free to ping us on Discord anytime. ❤️

**Comment by nasainsbury** [comment]
Thanks @chartgerink, committed your ideas, and will merge :)

**Pull request merged**